lib/sourcebit.js

```diff
@@ -1,22 +1,27 @@
+const { cloneDeep } = require('lodash');
 const debug = require('debug');
+const { diff: generateDiff } = require('deep-diff');
 const fs = require('fs');
 const mkdirp = require('mkdirp');
 const ora = require('ora');
 const path = require('path');
-const { cloneDeep } = require('lodash');
+const util = require('util');
+
 const { writeFrontmatterMarkdown, writeJSON, writeYAML } = require('./file-writers');
 
-const FILE_WRITERS = {
+const fileWriters = {
     'frontmatter-md': writeFrontmatterMarkdown,
     json: writeJSON,
     yml: writeYAML
 };
```

Comment:
@eduardoboucas you can use this battle-tested utility `stringifyDataByFilePath` to write all sort of files:
https://github.com/stackbithq/utils/blob/d1222b63ede765fb895668b641cb3f44c238d6ea/src/index.js#L373-L394
It takes into account several quirks that will prevent the app from crashing and store files safely.

Reply:
I'd rather not go through the trouble of changing the implementation unless something needs fixing. The code you linked looks good, but we have slightly different requirements (e.g. we don't need to choose the writer based on the file path, that is provided to us by the user). Also we're using a different YAML package, which a different syntax.

So if you have any concerns about a specific part of the existing implementation, can you flag those specifically?

Reply:
Ok, then just look at the YAML implementation in the link I've sent you and add `noRefs` attribute to make sure that if an object has circular references it won't be split. Also what about supporting TOML and markdown files with TOML and YAML frontmatters? I know Hugo and 11ty make use of these.